test/unit/basicio.cpp

```diff
@@ -83,6 +83,77 @@ TEST(basicio, output) {
   }
 }
 
+TEST(basicio, count_words_in_string) {
+    EXPECT_EQ(count_words_in_string("one two three four a"s), 5);
+    EXPECT_EQ(count_words_in_string(""s), 0);
+}
+
+TEST(basicio, get_dims) {
+    auto file = safe_open_for_reading("txt/matrix.txt");
```

Comment:
Se bolje bi bilo, ce bi bil vsak test v svojem direktoriju. To je se zlasti pomembno, ker nekateri testi tudi pisejo (in tega bo v prihodnje se vec), kar lahko vodi v probleme, ce se testi poganjajo vzporedno (na spinonu jih jaz pogosto pozenem kar z -j32).

Reply:
bi bilo dovolj, da so direktoriji kam pišejo za vsak test posebej, ali je vseeno boljše da so vsi testi posebej v svojih direktorijih?

Reply:
in kaj je -j32?

Reply:
Vsak test v svojem direktoriju. Tako kot v test/c++.
Z "ctest -j32" pozenes teste tako, da jih lahko do 32 tece socasno. Potem gre veliko hitreje.